File: user/src/main/java/com/kjj/user/controller/user/UserApiController.java
package com.kjj.user.controller.user;

import com.kjj.user.dto.user.UserInfoDto;
import com.kjj.user.service.user.UserService;
import jakarta.servlet.http.HttpServletRequest;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RestController
@RequestMapping("/v1/user")
public class UserApiController {

    private final UserService userService;

    @GetMapping("/info")
    public ResponseEntity<UserInfoDto> getInfo(@RequestParam("userId") Long id) {

        return ResponseEntity.ok(userService.getInfo(id));
    }
}

File: user/src/main/kotlin/com/kjj/user/controller/user/UserApiController.kt
package com.kjj.user.controller.user

import com.kjj.user.dto.info.UserInfoDto
import com.kjj.user.service.user.UserService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/user")
class UserApiController(
    val userService: UserService
) {
    @GetMapping("/info")
    fun getInfo(@RequestParam("userId") id: Long): ResponseEntity<UserInfoDto> {
        return ResponseEntity.ok(userService.getInfo(id))
    }
}